File: 2021/d1/python/d1p1_solution.py
# Advent of Code - Day 1 Problem 1

inputFile = open ('../input/d1_input.txt','r')

Lines = inputFile.readlines()
count = 0

# Iterrate through the lines and count which lines increment from the previous line
for i in range (len(Lines)-1):
    if int(Lines[i]) < int(Lines[i + 1]):
        count+=1

print(count)

inputFile.close()

File: 2021/d1/java/d1p1_solution.java
import java.io.*;
import java.util.*;

public class d1p1_solution {
    public static void main(String[] args) {
        int count = 0;
        int prev = 0;

        try {
            File inputFile = new File("../input/d1_input.txt");
            Scanner inputFileScanner = new Scanner(inputFile);

            // Read while the file has a next line
            while (inputFileScanner.hasNextLine()) {
                int depth = Integer.parseInt(inputFileScanner.nextLine());

                if (prev != 0 && depth > prev)
                    count += 1;

                prev = depth;
            }

            System.out.println(count);

            inputFileScanner.close();
        } catch (IOException e) {
            System.out.println(e);
        }
    }
}